libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -0,0 +1,614 @@
+              (unsigned)_msg_buff[_msg_buff_data_start+9],      // firmware major version
+              (unsigned)_msg_buff[_msg_buff_data_start+10],     // firmware minor version
+              (unsigned)_msg_buff[_msg_buff_data_start+11]);    // firmware revision
+        break;
+    }
+
+    case SiyiCommandId::HARDWARE_ID:
+        // unsupported
+        break;
+
+    case SiyiCommandId::AUTO_FOCUS:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        debug("AutoFocus:%u", (unsigned)_msg_buff[_msg_buff_data_start]);
+        break;
+
+    case SiyiCommandId::MANUAL_ZOOM_AND_AUTO_FOCUS: {
+        if (_parsed_msg.data_bytes_received != 2) {
+            unexpected_len = true;
+            break;
+        }
+        const double zoom_mult = UINT16_VALUE(_msg_buff[_msg_buff_data_start+1], _msg_buff[_msg_buff_data_start]) * 0.1;
+        debug("ZoomMult:%4.1f", (double)zoom_mult);
+        break;
+    }
+
+    case SiyiCommandId::MANUAL_FOCUS:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        debug("ManualFocus:%u", (unsigned)_msg_buff[_msg_buff_data_start]);
+        break;
+
+    case SiyiCommandId::GIMBAL_ROTATION:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        debug("GimbRot:%u", (unsigned)_msg_buff[_msg_buff_data_start]);
+        break;
+
+    case SiyiCommandId::CENTER:
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        debug("Center:%u", (unsigned)_msg_buff[_msg_buff_data_start]);
+        break;
+
+    case SiyiCommandId::ACQUIRE_GIMBAL_CONFIG_INFO: {
+        if (_parsed_msg.data_bytes_received != 5) {
+            unexpected_len = true;
+            break;
+        }
+        // update recording state and warn user of mismatch
+        const bool recording = _msg_buff[_msg_buff_data_start+3] > 0;
+        if (recording != _last_record_video) {
+            gcs().send_text(MAV_SEVERITY_ERROR, "Siyi: recording %s", recording ? "ON" : "OFF");
+        }
+        _last_record_video = recording;
+        debug("GimConf hdr:%u rec:%u foll:%u", (unsigned)_msg_buff[_msg_buff_data_start+1],
+                                               (unsigned)_msg_buff[_msg_buff_data_start+3],
+                                               (unsigned)_msg_buff[_msg_buff_data_start+4]);
+        break;
+    }
+
+    case SiyiCommandId::FUNCTION_FEEDBACK_INFO: {
+        if (_parsed_msg.data_bytes_received != 1) {
+            unexpected_len = true;
+            break;
+        }
+        const uint8_t fund_feedback_info = _msg_buff[_msg_buff_data_start];
+        const char* err_prefix = "Mount: Siyi";
+        switch (fund_feedback_info) {
+        case 0:
+            debug("FnFeedB success");
+            break;
+        case 1:
```

Comment:
presuambly an enumeration would be a good idea here.

Reply:
ok enum added.